webapp/channels/src/components/channel_threads_rhs/index.ts

```diff
@@ -0,0 +1,118 @@
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/common';
+import {isCollapsedThreadsEnabled} from 'mattermost-redux/selectors/entities/preferences';
+import {getCurrentTeam, getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
+import {makeGetThreadCountsInChannelView, makeGetThreadsInChannelView} from 'mattermost-redux/selectors/entities/threads';
+
+import {setGlobalItem} from 'actions/storage';
+import {closeRightHandSide, goBack, selectPostFromRightHandSideSearchByPostId, toggleRhsExpanded} from 'actions/views/rhs';
+import {getIsRhsExpanded, getPreviousRhsState} from 'selectors/rhs';
+import {getGlobalItem} from 'selectors/storage';
+import {StoragePrefixes} from 'utils/constants';
+
+import type {DispatchFunc, GetStateFunc} from 'mattermost-redux/types/actions';
+import {FetchChannelThreadOptions, FetchChannelThreadFilters} from '@mattermost/types/client4';
+import type {Channel} from '@mattermost/types/channels';
+import type {GlobalState} from 'types/store';
+
+import ChannelThreads, {Props, Tabs} from './channel_threads_rhs';
+
+function setChannelThreadsTab(tab: Tabs) {
+    return (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const teamId = getCurrentTeamId(getState());
+        const key = StoragePrefixes.CHANNEL_THREADS_TAB + teamId;
+        return dispatch(setGlobalItem(key, tab));
+    };
+}
+
+function getChannelThreadsTab(state: GlobalState) {
+    const teamId = getCurrentTeamId(state);
+    const isCRT = isCollapsedThreadsEnabled(state);
+    const tab = getGlobalItem(state, StoragePrefixes.CHANNEL_THREADS_TAB + teamId, Tabs.ALL);
+
+    if (!isCRT && tab === Tabs.FOLLOWING) {
+        return Tabs.ALL;
+    }
+
+    return tab;
+}
+
+function getThreadsForChannelWithFilter(channelId: Channel['id'], tab: Tabs, options?: FetchChannelThreadOptions) {
+    let filter;
+    if (tab === Tabs.FOLLOWING) {
+        filter = FetchChannelThreadFilters.FOLLOWING;
+    }
+    if (tab === Tabs.USER) {
+        filter = FetchChannelThreadFilters.USER;
+    }
+
+    return getThreadsForChannel(channelId, {...options, filter});
+}
+
+function makeMapStateToProps() {
+    const getThreadsInChannelView = makeGetThreadsInChannelView();
+    const getFollowingThreadsInChannelView = makeGetThreadsInChannelView('following');
+    const getCreatedThreadsInChannelView = makeGetThreadsInChannelView('created');
+    const getThreadCountInChannelView = makeGetThreadCountsInChannelView();
+
+    return (state: GlobalState) => {
+        const channel = getCurrentChannel(state);
+        const team = getCurrentTeam(state);
+        const currentUserId = getCurrentUserId(state);
+        const prevRhsState = getPreviousRhsState(state);
+
+        if (!channel || !team) {
+            return {
+                channel: {} as Channel,
+                canGoBack: false,
+                threads: [],
+                total: 0,
+            } as unknown as Props;
+        }
+
```

Comment:
I think `sidebar_right` has a loading screen for current team and current channel (`teamNeeded`, `currentChannelNeeded`), so I think this early return could be removed as this is downstream from that gate. `channel_info_rhs`'s index similarly depends on current channel but is safe because of the loading gate.

But if this is needed, 1. `as unknown as ...` seems like a hammer IMO and can potentially lead to issues and 2. would `threads` need to be `Props['all']`? (I think the forced casting might be hiding that typo if so)

```suggestion
```

Reply:
You're correct it's not needed!